evm/vm/forks/frontier/state.py

```diff
@@ -143,42 +154,44 @@ def run_post_computation(self, transaction, computation):
         gas_refund_amount = (gas_refund + gas_remaining) * transaction.gas_price
 
         if gas_refund_amount:
-            self.logger.debug(
+            self.vm_state.logger.debug(
                 'TRANSACTION REFUND: %s -> %s',
                 gas_refund_amount,
                 encode_hex(computation.msg.sender),
             )
 
-            self.account_db.delta_balance(computation.msg.sender, gas_refund_amount)
+            self.vm_state.account_db.delta_balance(computation.msg.sender, gas_refund_amount)
 
         # Miner Fees
-        transaction_fee = (transaction.gas - gas_remaining - gas_refund) * transaction.gas_price
-        self.logger.debug(
+        transaction_fee = \
+            (transaction.gas - gas_remaining - gas_refund) * transaction.gas_price
+        self.vm_state.logger.debug(
             'TRANSACTION FEE: %s -> %s',
             transaction_fee,
-            encode_hex(self.coinbase),
+            encode_hex(self.vm_state.coinbase),
         )
-        self.account_db.delta_balance(self.coinbase, transaction_fee)
+        self.vm_state.account_db.delta_balance(self.vm_state.coinbase, transaction_fee)
 
         # Process Self Destructs
         for account, beneficiary in computation.get_accounts_for_deletion():
             # TODO: need to figure out how we prevent multiple selfdestructs from
             # the same account and if this is the right place to put this.
-            self.logger.debug('DELETING ACCOUNT: %s', encode_hex(account))
+            self.vm_state.logger.debug('DELETING ACCOUNT: %s', encode_hex(account))
 
             # TODO: this balance setting is likely superflous and can be
             # removed since `delete_account` does this.
-            self.account_db.set_balance(account, 0)
-            self.account_db.delete_account(account)
+            self.vm_state.account_db.set_balance(account, 0)
+            self.vm_state.account_db.delete_account(account)
 
         return computation
 
 
-class FrontierState(BaseState, FrontierTransactionExecutor):
+class FrontierState(BaseState):
```

Comment:
I like this a lot more.  The inheritance piece there felt odd, adding extra methods to the `State`.